firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/Serial/Backend/Backend.tpp

```diff
@@ -160,11 +160,15 @@ Backend::Status Backend::input(uint8_t new_byte) {
       return Status::waiting;
   }
 
+  if (!states_.should_input(message.payload.tag)) {
```

Comment:
Looks good. Now that we're explicitly treating `States` as nothing more than a store, I think it might actually make sense to move the `should_input` method into the `Backend` class (and renaming the method to `Backend::accept_message`), as the logic inside `should_input` is actually about the behavior of `BackendReceiver`. If this sounds obvious to you, can you go ahead and implement it? Otherwise, let's first spend a bit of time discussing to see what ramifications of this change I might have forgotten about

Reply:
I think it's a straightforward change, I pushed it just now @ethanjli please review it.